client/web/src/hooks/auth.ts

```diff
@@ -18,6 +19,14 @@ export default function useAuth() {
     return apiFetch(url, "GET")
       .then((r) => r.json())
       .then((d) => {
+        if ((d as AuthResponse).synoAuth) {
+          fetch("/webman/login.cgi")
+            .then((r) => r.json())
+            .then((data) => {
+              setSynoToken(data.SynoToken)
```

Comment:
I think you could call `setSynoToken` from the same place that `updateCsrfToken` gets called. So it automatically picks it up on any request?

Reply:
That's too late.  The unraid CSRF token is only needed for POST requests, so the initial GET request to /api/data will succeed, and that will include the token.  For synology, the initial GET request will fail unless the SynoToken is present.  That's part of this strategy of calling /api/auth very early and getting all of the auth state setup before ever calling any other api endpoints.

Reply:
That said, we can (and should) actually move setting the unraid CSRF token to this initial /api/auth call, rather than it happening in the later /api/data call.

Reply:
I actually mean the `updateCsrfToken` call though, not the unraid one. So if it's present it automatically starts getting filled. And if not, it'll get filled when auth completes.

Reply:
FWIW, I think it's okay here too, I just think we'll also want it from the apifetch `.then` as well.

Reply:
oh, oops... I misread that.  The difference here is that there is potentially a CSRF token on every response.  For the SynoToken, it's only the authResponse that indicates should go fetch a synology token.

We could have a special error response that gets returned from all api calls which indicates that synology (or other) auth is needed.  Then we'd need to check for that error type, do whatever auth is wants, and then retry the request.  That's all possible, but I was trying to avoid the complexity of that by doing all of the auth up front.  The main counter-argument is if the synotoken eventually expires (I'm honestly not sure).  If that's the case, then you will eventually have requests start failing, and you'd need to reload the page to get a new synology token.

Reply:
as discussed in call, it appears that SynoToken is just a synology session ID, so just setting it up front and using it for subsequent requests should be fine.  This also has the added benefit that if for some reason the session expires or something, simply reloading the page will cause a new SynoToken to be fetched.

We don't think we should need to add auth failed responses  with retry logic on every `/api/*` call, but can add that later if it proves necessary.